src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala

```diff
@@ -538,6 +548,11 @@ trait TypeDiagnostics extends splain.SplainDiagnostics {
 
     override def traverse(t: Tree): Unit = {
       val sym = t.symbol
+      if (sym != null && sym.pos.isDefined)
+        sym.getAnnotation(UnusedClass).foreach { annot =>
+          if (!runReporting.suppressionExists(annot.pos))
+            runReporting.addSuppression(Suppression(annot.pos, List(MessageFilter.Category(WarningCategory.Unused)), sym.pos.start, sym.pos.end, synthetic = true))
```

Comment:
I'm not sure how reliable it is to use `sym.pos` here...

```
➜ sandbox qs -Wunused
Welcome to Scala 2.13.11-20221222-071937-84bf5ca (OpenJDK 64-Bit Server VM, Java 17.0.1).
Type in expressions for evaluation. Or try :help.

scala> object t { @annotation.unused private var x = 2 }
java.lang.UnsupportedOperationException: Position.point on NoPosition
	at scala.reflect.internal.util.Position.fail(Position.scala:24)
```

See `rangeFinder` in Typers

Reply:
I had thrown in `sym.pos.isDefined` for what I thought was good measure.